app/src/main/java/gholzrib/arctouchchallenge/ui/activities/MovieDetails.java

```diff
@@ -0,0 +1,182 @@
+package gholzrib.arctouchchallenge.ui.activities;
+
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.text.SpannableString;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.squareup.picasso.Callback;
+import com.squareup.picasso.Picasso;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Locale;
+
+import gholzrib.arctouchchallenge.R;
+import gholzrib.arctouchchallenge.core.models.Movie;
+import gholzrib.arctouchchallenge.core.models.TMDBConfiguration;
+import gholzrib.arctouchchallenge.core.utils.CheckConnection;
+import gholzrib.arctouchchallenge.core.utils.Constants;
+import gholzrib.arctouchchallenge.core.utils.PreferencesManager;
+
+import static gholzrib.arctouchchallenge.core.models.TMDBConfiguration.IMAGE_TYPE_BACKDROP;
+import static gholzrib.arctouchchallenge.core.models.TMDBConfiguration.IMAGE_TYPE_POSTER;
+
+public class MovieDetails extends AppCompatActivity implements View.OnClickListener {
+
+    private SimpleDateFormat appSdf = new SimpleDateFormat("dd/MM/yyyy", Locale.ROOT);
+    private SimpleDateFormat apiSdf = new SimpleDateFormat("yyyy-MM-dd", Locale.ROOT);
+
+    private int LNR_GENRES_WIDTH_LIMIT = 0;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_movie_details);
+
+        ImageView mBtnClose = (ImageView) findViewById(R.id.act_movie_details_btn_close);
+        RelativeLayout mRltPosterContainer = (RelativeLayout) findViewById(R.id.act_movie_details_rlt_poster_container);
+        ImageView mIvPoster = (ImageView) findViewById(R.id.act_movie_details_iv_poster);
+        final ProgressBar mPrgLoading = (ProgressBar) findViewById(R.id.act_movie_details_prg_loading);
```

Comment:
I only set the ProgressBar as final because it's required by the Callback used withe the Picasso library.